.travis.yml

```diff
@@ -16,7 +16,15 @@ install:
   - npm ci
 
 script:
-  - npm run-script build
+    if [[ "$TRAVIS_TAG" =~ "$PROD_TAG_REGEXP" ]]; then
```

Comment:
Heh, I actually found that exact same article when working on this! :-)

Even though I've formatted the command it over multiple lines for legibility, I do have all the required semi-colons in place for this to run correctly in a shell when it gets turned into a single line. (After a few earlier failed attempts, I "tested" my command in a shell before the most recent commit)

Judging by the Travis logs, it appears to now run as intended: https://travis-ci.org/buildit/buildit/builds/362533890?utm_source=github_status&utm_medium=notification

Reply:
It runs as intended, but the issue will arise if someone adds to any of the conditions to do another step.  In that case, it will execute the commands but if one fails and another passes, then it will still look like the build passed, because the If will be able to complete and return a pass code to travis.
I'll leave it up to you, the file will be rewritten anyway, so maybe it's fine to leave it until then.

Reply:
Probably best to tidy it now. You never know how long this stuff will survive before we get a chance to do that rewrite :-)

What do you recommend then? Going back to a list of individual `if [...]; then npm run whatever; fi` commands. If so, I have 2 questions:

* Doesn't Travis execute all of them - in which case, could that mess up the overall exit code and build status?
* For the last, catch-all one which currently uses `else`, should I replace it with a lengthy `if` that is the inverse of all the other conditions above it?

Or, should I use the YAML verbatim `|` operator as suggested in the StackOverflow article you linked to. That way, if someone extends the code in future but forgets a semi-colon it won't break. This seems like the nicer (and easier!) solution to me. What do you think?

Reply:
I've gone ahead and tried the `|` thing.

Reply:
👍

Reply:
😄 
In that case, would you mind adding your approval to this PR?

I'll poke @franz899 for feedback on it too in the mean time. Cheers!

Reply:
Fine by me. Reading thought the documentation is recommended to move the scripts into an external file: https://docs.travis-ci.com/user/customizing-the-build/#Implementing-Complex-Build-Steps but for now there is no need.